### ISSUE TITLE
IDE0062 false positive when a local function calls another non-static local function

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#46858]
**Version Used**: Visual Studio Version 16.8.0 Preview 1.0

**Steps to Reproduce**:

````c#
        public static void Main()
        {
            bool isValid = false;
            A();
            Console.WriteLine(isValid);

            // IDE0062 local function can be made static 
            // => static void A() is not valid because B is not static
            void A() => B();

            void B()
            {
                // B cannot be static as it use isValid
                isValid = true;
            }
        }
````

Look similar to #40259 but there is no struct parameter implied in this case. Feel free to close the issue if you consider this is a duplicate.

**Expected Behavior**: No diagnostic reported

**Actual Behavior**: diagnostic is reported
